Mathlib/Data/Matrix/Invertible.lean

```diff
@@ -53,6 +63,63 @@ protected theorem mul_mul_invOf_self_cancel (A : Matrix m n α) (B : Matrix n n
 #align matrix.invertible_of_invertible_mul invertibleOfInvertibleMul
 #align matrix.invertible_of_mul_invertible invertibleOfMulInvertible
 
+section conj_transpose
+variable [StarRing α] (A : Matrix n n α)
+
+/-- The conjugate transpose of an invertible matrix is invertible. -/
+instance invertibleConjTranspose [Invertible A] : Invertible Aᴴ := Invertible.star _
+
+lemma conjTranspose_invOf [Invertible A] [Invertible Aᴴ] : (⅟A)ᴴ = ⅟(Aᴴ) := star_invOf _
+
+/-- A matrix is invertible if the conjugate transpose is invertible. -/
+def invertibleOfInvertibleConjTranspose [Invertible Aᴴ] : Invertible A := by
+  rw [← conjTranspose_conjTranspose A, ← star_eq_conjTranspose]
+  infer_instance
+#align matrix.invertible_of_invertible_conj_transpose Matrix.invertibleOfInvertibleConjTranspose
+
+@[simp] lemma isUnit_conjTranspose : IsUnit Aᴴ ↔ IsUnit A := isUnit_star
+
+end conj_transpose
+
+end Semiring
+
+section CommSemiring
+
+variable [CommSemiring α] (A : Matrix n n α)
+
+/-- The transpose of an invertible matrix is invertible. -/
+instance invertibleTranspose [Invertible A] : Invertible Aᵀ where
+  invOf := (⅟A)ᵀ
+  invOf_mul_self := by rw [←transpose_mul, mul_invOf_self, transpose_one]
+  mul_invOf_self := by rw [←transpose_mul, invOf_mul_self, transpose_one]
+#align matrix.invertible_transpose Matrix.invertibleTranspose
+
+lemma transpose_invOf [Invertible A] [Invertible Aᵀ] : (⅟A)ᵀ = ⅟(Aᵀ) := by
+  letI := invertibleTranspose A
+  convert (rfl : _ = ⅟(Aᵀ))
+
+/-- `Aᵀ` is invertible when `A` is. -/
+def invertibleOfInvertibleTranspose [Invertible Aᵀ] : Invertible A where
+  invOf := (⅟(Aᵀ))ᵀ
+  invOf_mul_self := by rw [←transpose_one, ← mul_invOf_self Aᵀ, transpose_mul, transpose_transpose]
+  mul_invOf_self := by rw [←transpose_one, ← invOf_mul_self Aᵀ, transpose_mul, transpose_transpose]
+#align matrix.invertible__of_invertible_transpose Matrix.invertibleOfInvertibleTranspose
+
+/-- Together `Matrix.invertibleTranspose` and `Matrix.invertibleOfInvertibleTranspose` form an
+equivalence, although both sides of the equiv are subsingleton anyway. -/
+@[simps]
+def transposeInvertibleEquivInvertible : Invertible Aᵀ ≃ Invertible A where
```

Comment:
I'm not sure about this equiv. Will it ever be used except for your proof of `isUnit_transpose` below?

Reply:
Probably not, but this result is one that we already had in the other file, and I've somewhat established this pattern across the matrix inversion bits of the library, even though I admit it's not a very nice one.